### ISSUE TITLE
ValueError on compile when stateful

### ISSUE THREAD

**Issue by ekerazha** [fchollet/keras#1141]
After the fix for issue #1125 and #1130, I'm still trying to use a stateful RNN. This is my code

    in_out_neurons = 1
    hidden_neurons = 10
    batch_size = 10
    
    model = Sequential()
    model.add(GRU(hidden_neurons, batch_input_shape=(batch_size, 5, in_out_neurons), return_sequences=True, stateful=True))
    model.add(Dropout(0.2))
    model.add(TimeDistributedDense(in_out_neurons))
    model.add(Activation("linear"))
    model.compile(loss="mean_squared_error", optimizer="rmsprop")

and I get this error on compile

    ValueError: setting an array element with a sequence.

This is the traceback

    model.compile(loss="mean_squared_error", optimizer="rmsprop")

    models.py", line 372, in compile
    self.y_train = self.get_output(train=True)

    containers.py", line 73, in get_output
    return self.layers[-1].get_output(train)

    core.py", line 517, in get_output
    X = self.get_input(train)

    core.py", line 107, in get_input
    return self.previous.get_output(train=train)

    core.py", line 797, in get_output
    X = self.get_input(train)

    core.py", line 107, in get_input
    return self.previous.get_output(train=train)

    core.py", line 493, in get_output
    X = self.get_input(train)

    core.py", line 107, in get_input
    return self.previous.get_output(train=train)

    recurrent.py", line 79, in get_output
    K.set_value(self.states[i], states[i])

    theano_backend.py", line 345, in set_value
    x.set_value(np.asarray(value, dtype=x.dtype))

    numeric.py", line 474, in asarray
    return array(a, dtype, copy=False, order=order)

Thanks

**Comment by mynameisfiber** [comment]
Seems to be happening to me as well... installed 0933147 and get the following:

```
In [7]: model = Sequential()

In [8]: model.add(LSTM(20, return_sequences=True, batch_input_shape=(5,1,26), stateful=True))

In [9]: model.compile(loss='categorical_crossentropy', optimizer='rmsprop')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-5f42276ad4f2> in <module>()
----> 1 model.compile(loss='categorical_crossentropy', optimizer='rmsprop')

/usr/local/lib/python3.4/dist-packages/keras/models.py in compile(self, optimizer, loss, class_mode, theano_mode)
    370         self.X_test = self.get_input(train=False)
    371
--> 372         self.y_train = self.get_output(train=True)
    373         self.y_test = self.get_output(train=False)
    374

/usr/local/lib/python3.4/dist-packages/keras/layers/containers.py in get_output(self, train)
     71
     72     def get_output(self, train=False):
---> 73         return self.layers[-1].get_output(train)
     74
     75     def set_input(self):

/usr/local/lib/python3.4/dist-packages/keras/layers/recurrent.py in get_output(self, train)
     77         if self.stateful:
     78             for i in range(len(states)):
---> 79                 K.set_value(self.states[i], states[i])
     80
     81         if self.return_sequences:

/usr/local/lib/python3.4/dist-packages/keras/backend/theano_backend.py in set_value(x, value)
    343
    344 def set_value(x, value):
--> 345     x.set_value(np.asarray(value, dtype=x.dtype))
    346
    347

/usr/local/lib/python3.4/dist-packages/numpy/core/numeric.py in asarray(a, dtype, order)
    472
    473     """
--> 474     return array(a, dtype, copy=False, order=order)
    475
    476 def asanyarray(a, dtype=None, order=None):

ValueError: setting an array element with a sequence.

In [10]: model.layers[0]
Out[10]: <keras.layers.recurrent.LSTM at 0x7f6628331550>

In [11]: model.layers[0].states
Out[11]: [<CudaNdarrayType(float32, matrix)>, <CudaNdarrayType(float32, matrix)>]
```

**Comment by ekerazha** [comment]
@mynameisfiber
Did you find a solution?

It seems like

    array(a, dtype, copy=False, order=order)

can't convert the theano tensor to a numpy array.

I found a similar issue here: https://stackoverflow.com/questions/23643850/how-to-swich-theano-tensor-to-numpy-array

Unfortunately, I'm not able to fix the code. @fchollet? @EderSantana?

**Comment by ekerazha** [comment]
This particular issue has been fixed with commit 07ffc76.
